R/deployApp.R

```diff
@@ -611,22 +611,22 @@ openURL <- function(client, application, launch.browser, on.failure, deploymentS
     # or open no url if things failed
 }
 
-runStartupScripts <- function(appDir, logLevel) {
+runStartupScripts <- function(appDir, quiet = FALSE, verbose = FALSE) {
   scripts <- c(
     # the site-wide startup script
     file.path(R.home("etc"), "rsconnect.site"),
     # the user startup script
     path.expand("~/.rsconnect_profile"),
     # a startup script specific to this application
-    file.path(appDir, ".rsconnect_profile"))
+    file.path(appDir, ".rsconnect_profile")
+  )
+  scripts <- scripts[file.exists(scripts)]
 
   # iterate over the startup scripts
   for (script in scripts) {
-    if (file.exists(script)) {
-      if (logLevel == "verbose") {
-        cat("----- Sourcing startup script", script, "-----\n")
-      }
-      source(file = script, verbose = (logLevel == "verbose"))
-    }
+    taskStart(quiet, "Running {script}")
+
+    env <- new_environment(parent = globalenv())
+    source(script, verbose = verbose, local = env)
```

Comment:
Idle musing: This isn't a current use case, but we could eventually have a stable environment across all of these user tasks, which might let folks (somehow) build state during the "pre" step that can then get consumed during the "post" step..